### ISSUE TITLE
Proposal for further development

### ISSUE THREAD

**Issue by roland-KA** [DannySeidel/typst-dhbw-template#20]
Hi @DannySeidel, I'm following the development of Typst since the first beta release in Spring 23, but I really haven't used it up to now (just a bit playing around). Recently a few of my students asked, it they can use Typst for their "Studienarbeit" and I agreed and proposed them to use your template.

This was the reason for me, to have a closer look on Typst (again) as well as on your template, which is by the way a great piece of work! Apart from that I think Typst itself is now mature enough to be used for "real work". So I came to the conclusion that I would like to recommend Typst and your template in the future to my students for their bachelor thesis, Studienarbeiten etc.

Furthermore I came to the point, where I thought, that this would also be a good occasion to improve the DHBW template in general for these sorts of documents. I am sort of an end-user for these documents as I have to read them and work with them on a regular basis. And to be honest, I'm not quite satisfied with the "usability" of what I typically get. Therefore my goal is to 
- reduce "clutter and noise" and make the document "cleaner" and more usable
- improve legibility by following professional typographic recommendations

thus making sort of a "simplified-DHBW" template.

What does that mean in detail? 

Well, the structure, typography and layout of a document in this context should convey the content to the reader as simply as possible. And one example where this is not the case at the moment is the vast amount of content between the title page and the first chapter. As a user I want to get to the content in the first chapter as fast as possible. But currently after opening a document, I have to scroll through a ton of pages which are completely useless to me. By the way: Do you know anybody who really ever uses the list-of-tables, the list-of-figures or anything like this? ... I don't 🤓. So one of the changes, I have in mind would be to switch off all (or most) of that stuff by default and if it gets switched on, to place it at the end of the document in the appendix. That's anyway the place in templates of other universities where elements like the glossary, the list of acronyms, the declaration of authorship etc. are put.

An example for following typographic recommendations would be to reduce line length, as a line should not contain more than 72 letters for optimal legibility (rather less; in some books on Typography only 60 letters are proposed).

I've also created a version of the title page that is sort of "cleaner" to illustrate my ideas (see image below). I've made the following changes to the current version:
- In the text I eliminated all the prepositions (like "im", "von" etc.); they bear no real information
- There are now three distinct blocks, each using a uniform line height 
  - Title, 2em
  - Subtitles, 1,5em
  - Information block, 1em (which contains now also the date)
 - Only one type of alignment: All text is now centered

<img width="676" alt="image" src="https://github.com/user-attachments/assets/82a5def7-e444-4f3a-b01b-9251272fcb6c">

These are of course just first ideas which have to be discussed in further detail. And some things might be feasible, others not.

But my main question to you @DannySeidel  is now: Would you support such further development? I would like to avoid a fork of your template so that we retain exactly one DHBW template in the Typst-world. But to do that, I need of course first your agreement and then your support for discussing changes, reviewing my PRs etc. 

What do you think about all this? 😊

**Comment by roland-KA** [comment]
Hi @DannySeidel. did you make up your mind, if you would like to support the proposed development? Of course I can fully understand, if you don't want to change the template that much and keep it as is. It would just be helpful for me to know your opinion, so that I can proceed in some direction.

**Comment by DannySeidel** [comment]
Hi @roland-KA,

I haven't had much time lately, which is why the reply took so long. First of all, thanks for the long proposal. I first found out about Typst in the spring of this year. I took a look and instantly decided to use it for my remaining theses at DHBW because everything was so much simpler than in LaTeX. The only problem was that there was no template that met all my expectations, so I decided to create my own.

I have now used Typst together with my template to write my "Studienarbeit," Bachelor thesis, and to create documentation for group projects in lectures. From that experience, I have to say that it was sufficient for everything I wanted to do.

Regarding your proposal, I'm generally always open to improvement ideas, although I think the template is in pretty good shape by now. Concerning the structuring of the document, I mostly try to conform to the existing DHBW guidelines, which you can find here. Looking at these guidelines, they propose including all the directories in the preface of the document (you can find that in `8.2 Gliederung der Arbeit` on page 18). I never really thought about the person reading the documents because I was always the one writing the document, but I agree that some are not really useful. I think that at least the table of contents and the acronyms would be useful to me. As I was already saying, I try to conform to the existing guidelines and would therefore like to keep this as it is now.

Now, taking a look at the title page, I like to give the end user a lot of flexibility while keeping a somewhat general structure. The main point behind aligning the bottom content to the left is to have more flexibility. First, the template can also be used by more than one author, and second, if you have a company with a long name, the company may not fit on one line. If you use the template, you often don't see how many different layouts are possible on the title page. To guarantee that all possible layouts look good, I aligned the content on the left as this made it easier. In the end, I think it is somewhat a personal preference where the content is aligned.

Lastly, about the prepositions. I'm open to removing them if you think that helps make the template more concise, although I think that's more about personal preference again. I don't really have an opinion for or against it. I would like to have a few other opinions on that.

In general, I'm not really sure what the best way is to contribute to a Typst template. Of course, you can create a PR in this repo, but this will not change the template on the official Typst website. Right now, I have a fork of the Typst packages repo from where I contribute. If I were to update something there, I would use this repo as a mirror and push a commit with the changes to this repo too. That's why the `readme` file of this repo says that you should rather create an issue instead of a PR because in the end, I still have to create a PR in the Typst packages repo.

To summarize, if you have ideas, you can always propose them here, but I don't want to change too much for now. As I already said, I think the template has reached a really good state in my opinion.

Thanks again for the ideas and for recommending my template. It's nice to see that I not only did all the work for myself 😊

**Comment by roland-KA** [comment]
Hi @DannySeidel,

> I have now used Typst together with my template to write my "Studienarbeit," Bachelor thesis, and to create documentation for group projects in lectures. From that experience, I have to say that it was sufficient for everything I wanted to do.

That reassures my plans to recommend Typst and your template to others!

> To summarize, if you have ideas, you can always propose them here, but I don't want to change too much for now. As I already said, I think the template has reached a really good state in my opinion.

That's also my impression of the template. You are covering quite a few use cases (and as I see from your comments, you've also dealt with several edge cases). Moreover you have built in quite a lot of flexibility. 

> the existing DHBW guidelines ... in 8.2 Gliederung der Arbeit on page 18

This is the thing I have issues with 🤓! I think these guidelines need some overhaul. Some aspects, like the recommendation to use a 1.5x line-height even stem from the times where typewriters were used.  On a typewriter you could only choose a line-height of 1x/1.5x/2x and 3x (1x being the standard). As professors wanted to write their comments in between the lines, there had to be more space. That's the reason for this recommendation! 

Nowadays however, people use the comment function of their favourite PDF reader  to achieve this. And with a line-height of 1.5x you will not obtain optimum legibility. Experts in typography (see e.g: Robert Bringhurst, _The elements of typographic style_, Hartley & Marks, 1997) recommend e.g. for a 11pt font a line-heigth of 13-15pt, depending on the font (_Verdana_ with its large glyphs needs more space than e.g. _Calibri_), length of line and other aspects. 

This is just one example. I've already mentioned other points which reduce the usability of the resulting documents in my proposal above. Fortunately the current DHBW guidelines are just a recommendation which gives way to change and improvement. So my plan is to design a Typst template that shows how a modernised document  (fit for the digital world) could look like. 

Within our discussion it became clearer to me, that this goal is different of what you were pursuing with the existing template. Therefore I think it is in the end better to leave the current template as is. Of course a few improvements here and there are always fine, but in general it has reached a mature state and is useful to many people. 

Therefore I will move on with a fork of your work and create on that basis another template. That gives me more degrees of freedom to try out new things as I don't have to consider some form of "backward compatibility". Nevertheless I would be happy to discuss from time to time some ideas with you; especially as you have more experience with Typst.

**Comment by DannySeidel** [comment]
Hi @roland-KA,

Thanks for the insights. As I said before, I don't like many of the guidelines either. I'm curious to see how your "improved" version will look.

I'm looking forward to discussing your ideas 🙂
